fdb-record-layer-lucene/src/test/java/com/apple/foundationdb/record/lucene/LuceneIndexTest.java

```diff
@@ -659,39 +712,46 @@ void repartitionGroupedTest() throws ExecutionException, InterruptedException, I
-                "text:propose",
-                new Sort(new SortField("timestamp", SortField.Type.LONG, true)),
-                1);
+    private void validateDocsInPartition(Index index, int partitionId, Tuple groupingKey, int expectedHitCount, Set<Tuple> expectedIds) throws IOException {
+        LuceneScanQuery scanQuery;
+        if (groupingKey.isEmpty()) {
+            scanQuery = (LuceneScanQuery) LuceneIndexTestUtils.fullSortTextSearch(recordStore, index, "text:propose", null);
+        } else {
+            scanQuery = (LuceneScanQuery) groupedSortedTextSearch(index,
+                    "text:propose",
+                    null,
+                    groupingKey.getLong(0));
+        }
+        IndexMaintainerState state = new IndexMaintainerState(recordStore, index, recordStore.getIndexMaintenanceFilter());
+        IndexReader indexReader = FDBDirectoryManager.getManager(state).getIndexReader(groupingKey, partitionId);
+        LuceneOptimizedIndexSearcher searcher = new LuceneOptimizedIndexSearcher(indexReader);
+        TopDocs newTopDocs = searcher.search(scanQuery.getQuery(), Integer.MAX_VALUE);
 
-        RecordCursor<IndexEntry> cursor = recordStore.scanIndex(COMPLEX_PARTITIONED,
-                scanQuery,
-                continuation,
-                ExecuteProperties.newBuilder().setSkip(amountToSkip).setReturnedRowLimit(1).build().asScanProperties(false));
+        assertNotNull(newTopDocs);
+        assertNotNull(newTopDocs.scoreDocs);
+        assertEquals(expectedHitCount, newTopDocs.scoreDocs.length);
 
-        List<IndexEntry> entries = cursor.asList().join();
-        assertEquals(1, entries.size());
-        RecordCursorResult<IndexEntry> lastResult = cursor.onNext().get();
-        assertEquals(RecordCursor.NoNextReason.RETURN_LIMIT_REACHED, lastResult.getNoNextReason());
-        LuceneContinuationProto.LuceneIndexContinuation parsed = LuceneContinuationProto.LuceneIndexContinuation.parseFrom(lastResult.getContinuation().toBytes());
-        // we stopped in the expected partition
-        assertEquals(expectedPartitionId, parsed.getPartitionId());
-        assertEquals(expectedDocId, entries.get(0).getPrimaryKey());
-        return lastResult.getContinuation().toBytes();
+        Set<String> fields = Sets.newHashSet(LuceneIndexMaintainer.PRIMARY_KEY_FIELD_NAME);
+        assertEquals(expectedIds, Arrays.stream(newTopDocs.scoreDocs).map(scoreDoc -> {
+            try {
+                Document document = searcher.doc(scoreDoc.doc, fields);
+                IndexableField primaryKey = document.getField(LuceneIndexMaintainer.PRIMARY_KEY_FIELD_NAME);
+                return Tuple.fromBytes(primaryKey.binaryValue().bytes);
+            } catch (IOException e) {
+                throw new RuntimeException(e);
+            }
+        }).collect(Collectors.toSet()));
     }
 
     @Test
-    void repartitionSyntheticGroupedTest() {
+    void repartitionSyntheticGroupedTest() throws IOException {
```

Comment:
This test could also benefit from using `validateDocsInPartition`

Reply:
I still think it would be good to add `validateDocsInPartition` here, but it might make sense to first add a version that asserts the constraints of partitions, rather than having to explicitly declare which documents are in which partition.